vars/googleCloudFunctionDeploy.groovy

```diff
@@ -103,64 +88,25 @@ def getGcloudPodYaml() {
     """
 }
 
-def copyArchive(String functionName) {
-    container('gcloud') {
-        script {
-            try {
-                def archiveName = "${functionName}.zip"
-                copyArtifacts(
-                    projectName: env.JOB_NAME, 
-                    selector: specific(env.BUILD_NUMBER), 
-                    filter: archiveName, 
-                    fingerprintArtifacts: true
-                )
-                
-                sh """
-                    if [ ! -f "${archiveName}" ]; then
-                        echo "Error: Archive file not found in current directory"
-                        exit 1
-                    fi
-                """
-            } catch (Exception e) {
-                error "Failed to process function archive: ${e.message}"
-            }
-        }
-    }
-}
-
-def uploadArchive(Map config) {
-    sh """
-        gcloud config set project ${config.projectId}
-        gcloud storage cp ${config.functionName}.zip ${config.gcsPath}
-    """
-}
-
 def deployFunction(Map config) {
-    def deployCommand = "gcloud functions deploy ${config.functionName}"
-    
-    if (config.generation == 'gen2') {
-        // Jenkins is very sensitive to indentation, so this snippet looks strange but it's correct
-        deployCommand += """ --gen2 \\
-        --concurrency ${config.concurrency ? config.concurrency : 1} \\
-        --cpu ${config.cpu ? config.cpu : 1} \\"""
-    } else {
-        // Have to explicitly set since gen2 is becoming the default
-        deployCommand += " --no-gen2 \\"
-    }
-    
+    def deployCommand = "gcloud functions deploy ${config.functionName} \\"
+
     deployCommand += """
+        --gen2 \\
+        --project ${config.projectId} \\
         --runtime ${config.runtime} \\
```

Comment:
So you decided to keep runtime; do you know if this will work on multiple runtimes? My intitial assumptions is "yes" but I'm curious if you tested this. Can't see there being an issue tho.

Reply:
Decided to leave this in as it doesn't hurt anything at the moment. But in particular I'm thinking about a unit test stage as a later addition which will be made a bit more cumbersome by leaving language open to config. But yeah I think we can leave it in for now and I've cross that bridge when I get there